File: Plugins/Aspose_Cells_Java_for_PHP/src/aspose/cells/WorkingWithFiles/UtilityFeatures/WorksheetToImage.php
<?php

namespace Aspose\Cells\WorkingWithFiles\UtilityFeatures;

use com\aspose\cells\Workbook as Workbook;
use com\aspose\cells\ImageFormat as ImageFormat;
use com\aspose\cells\ImageOrPrintOptions as ImageOrPrintOptions;
use com\aspose\cells\SheetRender as SheetRender;



class WorksheetToImage {

    public static function run($dataDir=null)
    {
        $imageFormat = new ImageFormat();
        //Instantiate a new workbook with path to an Excel file
        $book = new Workbook($dataDir . "MyTestBook1.xls");

        //Create an object for ImageOptions
        $imgOptions = new ImageOrPrintOptions();

        //Set the image type
        $imgOptions->setImageFormat($imageFormat->getPng());

        //Get the first worksheet.
        $sheet = $book->getWorksheets()->get(0);

        //Create a SheetRender object for the target sheet
        $sr = new SheetRender($sheet, $imgOptions);
        for ($j = 0; $j < $sr->getPageCount(); $j++)
        {
            //Generate an image for the worksheet
            $sr->toImage($j, $dataDir . "mysheetimg" . $j . ".png");

        }

        // Print message
        print "Images generated successfully." . PHP_EOL;
    }

}

File: Plugins/Aspose-Cells-Java-for-Jython/asposecells/WorkingWithFiles/WorksheetToImage.py
from asposecells import Settings
from com.aspose.cells import Workbook
from com.aspose.cells import ImageFormat
from com.aspose.cells import ImageOrPrintOptions
from com.aspose.cells import SheetRender




class WorksheetToImage:

    def __init__(self):
        dataDir = Settings.dataDir + 'WorkingWithFiles/WorksheetToImage/'
        
        imageFormat = ImageFormat
        
        #Instantiate a workbook with path to an Excel file
        book = Workbook(dataDir + "Book1.xls")

        #Create an object for ImageOptions
        imgOptions = ImageOrPrintOptions()

        #Set the image type
        imgOptions.setImageFormat(imageFormat.getPng())

        #Get the first worksheet.
        sheet = book.getWorksheets().get(0)

        #Create a SheetRender object for the target sheet
        sr =SheetRender(sheet, imgOptions)
        for i in range(sr.getPageCount()):
        
            #Generate an image for the worksheet
            sr.toImage(i, dataDir + "mysheetimg" + ".png")

        
        # Print message
        print "Images generated successfully."
 
if __name__ == '__main__':        
    WorksheetToImage()